EssentialFeed/EssentialFeedTests/Image Comments API/LoadImageCommentsFromRemoteUseCaseTests.swift

```diff
@@ -0,0 +1,192 @@
+//
+//  LoadImageCommentsFromRemoteUseCaseTests.swift
+//  EssentialFeedTests
+//
+//  Created by Bogdan Poplauschi on 27/03/2021.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import EssentialFeed
+import XCTest
+
+final class LoadImageCommentsFromRemoteUseCaseTests: XCTestCase {
+	func test_init_doesNotRequestDataFromURL() {
+		let (_, client) = makeSUT()
+		
+		XCTAssertTrue(client.requestedURLs.isEmpty)
+	}
+	
+	func test_load_requestsDataFromURL() {
+		let url = anyURL()
+		let (sut, client) = makeSUT(url: url)
+		
+		sut.load { _ in }
+		
+		XCTAssertEqual(client.requestedURLs, [url])
+	}
+	
+	func test_loadTwice_requestsDataFromURLTwice() {
+		let url = anyURL()
+		let (sut, client) = makeSUT(url: url)
+		
+		sut.load { _ in }
+		sut.load { _ in }
+		
+		XCTAssertEqual(client.requestedURLs, [url, url])
+	}
+	
+	func test_load_deliversErrorOnClientError() {
+		let (sut, client) = makeSUT()
+		
+		expect(sut, toCompleteWith: failure(.connectivity), when: {
+			client.complete(with: NSError(domain: "Test", code: 0))
+		})
+	}
+	
+	func test_load_deliversErrorOnNon200HTTPResponse() {
+		let (sut, client) = makeSUT()
+		let samples = [199, 201, 300, 400, 500]
+		
+		samples.enumerated().forEach { index, code in
+			expect(sut, toCompleteWith: failure(.invalidData), when: {
+				client.complete(withStatusCode: code, data: anyData(), at: index)
```

Comment:
This test should use a valid response to be able to test the `status code` logic.

But `anyData` is not a valid response for this endpoint based on API the contract. 

It could be an empty JSON list: 

```suggestion
				client.complete(withStatusCode: code, data: makeItemsJSON([]), at: index)
```